### ISSUE TITLE
All archs SIMDs are being built for libjpeg-turbo on Android builds

### ISSUE THREAD

**Issue by cocus** [glmark2/glmark2#144]
Hi,
I'm trying to build this under the Android build system for an aarch64 target. I've hit some other issues, but this one seems to be the most interesting.
I had some errors while running the suggested `build GLMark2`, pointing to "undefined instructions" errors on some mips assembler file.
After fixing (or rather, hacking) the Android.mk so only `*arm64*S` files are being included on `LOCAL_SRC_FILES`, I hit another issue.
This time ALL the .c files from the `simd/` directory were being added. Specifically the "altivec" ones; which are not supported on aarch64. I've also hacked this by just adding `/src/libjpeg-turbo/simd/jsimd_arm64.c` instead of ALL simd c files on the same `LOCAL_SRC_FILES` variable.
Since the sources are NOT architecture aware (maybe the original autoconfig files are), there should be some kind of logic on this Android.mk file that only adds architecture dependant sources; and not ALL files.

**Comment by afrantzis** [comment]
Hi! Are you following the instructions at the top of [INSTALL.android](https://github.com/glmark2/glmark2/blob/master/INSTALL.android) or using some different method? Running the `build.sh` script should end up using `jni/Android.ndk.mk` which builds a non-simd version of libjpeg-turbo, specifically to avoid the architecture issues you mention.

**Comment by cocus** [comment]
I actually ran the step under "Building using the Android build system", where I only had to run "make GLMark2".
Should I go for the instructions at the top?

**Comment by afrantzis** [comment]
@cocus Yes, the top instructions describe the currently supported way to build. We need to go through the android documentation to clean things up, since a lot has changed since these were first written.

Perhaps "make GLMark2" could also by simply fixed by replacing `jni/Android.mk` with `jni/Android.ndk.mk` in the android/Android.mk file.

**Comment by cocus** [comment]
I haven't tried sadly, I asked the vendor to try to compile this by themselves and I got the apk out of them. I think they mostly followed the top instructions. You can close this issue :)
(but I would like to see some changes in the docu)

Thanks!
